lib/Command/UpdateAccount.php

```diff
@@ -0,0 +1,154 @@
+<?php
+
+/**
+ * @author Christoph Wurst <christoph@winzerhof-wurst.at>
+ * @author Maadix
+ *
+ * Mail
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+namespace OCA\Mail\Command;
+
+use OCA\Mail\Db\MailAccountMapper;
+use OCP\Security\ICrypto;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class UpdateAccount extends Command {
+	public const ARGUMENT_USER_ID = 'user-id';
+	public const ARGUMENT_EMAIL = 'email';
+	public const ARGUMENT_IMAP_HOST = 'imap-host';
+	public const ARGUMENT_IMAP_PORT = 'imap-port';
+	public const ARGUMENT_IMAP_SSL_MODE = 'imap-ssl-mode';
+	public const ARGUMENT_IMAP_USER = 'imap-user';
+	public const ARGUMENT_IMAP_PASSWORD = 'imap-password';
+	public const ARGUMENT_SMTP_HOST = 'smtp-host';
+	public const ARGUMENT_SMTP_PORT = 'smtp-port';
+	public const ARGUMENT_SMTP_SSL_MODE = 'smtp-ssl-mode';
+	public const ARGUMENT_SMTP_USER = 'smtp-user';
+	public const ARGUMENT_SMTP_PASSWORD = 'smtp-password';
+
+	/** @var mapper */
+	private $mapper;
+
+	/** @var ICrypto */
+	private $crypto;
+
+	public function __construct(MailAccountMapper $mapper, ICrypto $crypto) {
+		parent::__construct();
+
+		$this->mapper = $mapper;
+		$this->crypto = $crypto;
+	}
+
+	/**
+	 * @return void
+	 */
+	protected function configure() {
+		$this->setName('mail:account:update');
+		$this->setDescription('Update a user\'s IMAP account');
+		$this->addArgument(self::ARGUMENT_USER_ID, InputArgument::REQUIRED);
+		$this->addArgument(self::ARGUMENT_EMAIL, InputArgument::REQUIRED);
```

Comment:
let's identify the account by the account id

Reply:
How do you imagine the use of the account id argument? I can think about an optional argument, but this make no sense if  user_id is still required. 
Using only  findById instead of findByUserId may not work in many environment. (This would be our case. We launch the command from an external mail service which does not know the NC user id)
One solution I can think of is keeping ARGUMENT_EMAIL as Required. Then user_id and account id as optional, checking that at least one of them is present.
Would this work for you?

Reply:
I would use the account ID over the email.

There is no guarantee that the email account is only set up for one single user. It could be there for many, if they share the account. I'm not sure if we want to update all copies in bulk. I would rather prefer finer control by the unique ID.

If you need this for scripting we could add a command that lists the account IDs for a given email.

Reply:
Do you mean that one unique mail account (with same id) can be assigned to multiple NC users that can access the same IMAP folders?
If not , I understand that the combination user_id + email is already ensuring that we are editing the correct mail account as there can not be two users with same user_id.

Reply:
Two users can add the same IMAP credentials. Those accounts with both have their unique ID. Only the email address isn't unique.

Reply:
> If you need this for scripting we could add a command that lists the account IDs for a given email.

If we want to avoid bulk updating, the command should return a unique account ID by the combination of username + email address.

If we agree on this, I am not able to see the difference between my approach and yours.
In any case, if you consider that others may  need to query by mail-address ID, no problem. What we would like to preserve is the ability to query the address ID by the combination of both values (username and mail address) to have a unique result.

Reply:
See the account sync/diagnose/export-threads/train/delete commands. They all use the account ID. The account ID is the primary key of the oc_mail_accounts table. The update command should use the same identifier.

I can finish the PR of your if the requirement isn't clear.

Reply:
Hi,
I have updated the code as you mentioned. Hope it's fine.
Please tell us if there is something missing